interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,1029 @@
+        Boolean flag to allow files to be overwritten or not
+    """
+    if not isinstance(solution, ct.Solution):
+        if not isinstance(solution, PurePath):
+            try:  # try to turn full path into pathlib Path
+                solution = Path(solution)
+            except:
+                raise ValueError(f"Invalid path: {str(solution)}")
+
+            if len(solution.parts) == 1:
+                solution = Path.cwd() / solution
+
+        try:
+            solution = ct.Solution(str(solution))
+        except:
+            raise ValueError(f"Invalid path: {str(solution)}")
+            
+    if mech_path is None:
+        mech_path = Path.cwd() / f"{solution.name}.ck"
+
+    # NOTE: solution.transport_model returns back a string that is None right now
+    if solution.transport_model == "None" or solution.transport_model is None:
+        tran_path = None
+    elif tran_path is None:
+        tran_path = f"{solution.name}.tran"       
+
+    ext = {'mech': 'ck', 'thermo': 'therm', 'tran': 'tran'}
+    path = {'mech': mech_path, 'thermo': thermo_path, 'tran': tran_path}
+    parent = Path.cwd()
+
+    for key, val in path.items():
+        if val is None:
+            continue
+
+        if not isinstance(val, PurePath):
+            # try to turn full str into pathlib Path
+            try:  
+                path[key] = Path(val)
+            except: 
+                raise ValueError("Could not make a path from: {val}")
+        
+        if key == 'mech':
+            # only want to reset parent to mech parent if it has parents
+            if len(path[key].parts) > 1:
+                # directory given
+                if not path[key].suffix:
+                    parent = path[key]
+                    path[key] = parent / f"{solution.name}.{ext[key]}"
+
+                # file given
+                else:
+                    parent = path[key].parent
+                    path[key] = parent / val
+
+        # if no extension was specified, use default
+        if not path[key].suffix: 
+            path[key] = parent / f"{path[key].name}.{ext[key]}"
+
+        # it's a file, but does it have a directory?
+        elif len(path[key].parts) == 1:
+            path[key] = parent / path[key]
+
+        # delete file if it exists or throw error
+        if path[key].is_file(): 
+            if overwrite:
+                path[key].unlink()
+
+            else:
+                raise Exception("Path already exists."\
+                                "\nSpecify a new path or set overwrite to True"\
+                               f"\nCurrent Path: {str(path[key])}")
+
+    # write mechanism file
+    text = []
+    text.append(Header(solution).build_text())
+    text.append(Species(solution).build_text())
+    if path['thermo'] is None:
+        text.append(Thermodynamics(solution.species(), separate_file=False).build_text())
+    text.append(Mechanism(solution, reorder_reactions=reorder_reactions).build_text())
```

Comment:
I don't think these high-level abstractions need to be classes. These might as well be called `build_x_text()` and just be a function. Having them as classes implies that they will hold some state that should be modified, or that there is potentially some useful inheritance behavior, but that's not the case here.